src/common/os/posix/os_utils.cpp

```diff
@@ -393,4 +395,147 @@ void CtrlCHandler::handler(void*)
 	terminated = true;
 }
 
+namespace {
+
+	class Converter
+	{
+	public:
+		virtual char* convert(const char* fromBuf, size_t fromLen) = 0;
+		virtual void destroy(char* &outBuf) = 0;
+	};
+
+	// No real conversion, system locale is UTF-8 or iconv is not available
+	class NullConverter : public Converter
+	{
+		char* convert(const char* fromBuf, size_t fromLen) override
+		{
+			return const_cast<char*>(fromBuf);
+		}
+		void destroy(char* &outBuf)
+		{
+			outBuf = NULL;
+		}
+	};
+
+#ifdef HAVE_ICONV_H
+	class IConverter : public Converter
+	{
+	public:
+
+		IConverter(const char* sysLocale) : ic(NULL)
```

Comment:
Class name "IConverter" points that class is (kind of) interface while this is not so.
Base class "Converter" is pure virtual, i.e. it is good candidate to be "IConverter" :)
Please, avoid confusion when choosing names.